sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/AQEShuffleReadExec.scala

```diff
@@ -200,7 +200,14 @@ case class AQEShuffleReadExec private(
       val numCoalescedPartitionsMetric = metrics("numCoalescedPartitions")
       val x = partitionSpecs.count(isCoalescedSpec)
       numCoalescedPartitionsMetric.set(x)
-      driverAccumUpdates += numCoalescedPartitionsMetric.id -> x
+      val numEmptyPartitionsMetric = metrics("numEmptyPartitions")
+      val y = child match {
+        case s: ShuffleQueryStageExec =>
```

Comment:
We may not reach it for canonicalized plans or other unknown cases. Shall we be conservative and add a `case _ => 0`?